### ISSUE TITLE
Editor aggressively deletes your game data from collections when you temporarily create a code error

### ISSUE THREAD

**Issue by j-h-a** [defold/defold#9682]
**Describe the bug**

The editor silently edits project files that are not open, removing overridden `component_property` values, when a code error in an externally edited script is discovered during an attempted build.

**To Reproduce**

In a project with collections that have overridden `go.property` values, edit the script that defines those values to create a code error that will refuse to build, then fix the error.

The editor thinks the script no longer declares its `go.property` values and aggressively removes overrides of those values from files on disk (including files that are not open).

**Expected behavior**

Editor does not delete user data.

Note: This behaviour is probably desired when `go.property` values are removed, or possibly renamed, but it would probably be better to design some different behaviour. Here is an initial proposal:
* Leave all values in place in project files - even if the property no longer exists.
* Ignore properties that don't exist when building/creating the runtime files, but leave them present in the files.
* Notify user with build warnings that explain there are overrides where the parent property no longer exists.
* Only when user opens the offending files, prompt them to keep existing values and rename the properties they override, or remove them.

**Defold version:**

1.9.4

**Platforms:**
 - Platforms: macOS
 - OS: Sonoma 14.5

**Workaround:**

Use source control to keep a close eye on changes and revert changes the editor incorrectly makes.

**Comment by j-h-a** [comment]
To help this get fixed, here's an incremental way to fix it in 2 or 3 stages:

* Leave all values in place in project files - even if the property no longer exists.
* Ignore properties that don't exist when building/creating the runtime files, but leave them present in the files.

The above two could be done first, which would prevent the problematic behaviour.

* Notify user with build warnings that explain there are overrides where the parent property no longer exists.

This could be done separately - although it might be quite simple so could be bundled with the other two.

* Only when user opens the offending files, prompt them to keep existing values and rename the properties they override, or remove them.

This might be a bit more tricky and could potentially be split out into a separate "nice-to-have" ticket.

----

A a *proposed* new behaviour, it would be good to get some feedback. I can suggest ways to split it but if it's not a good/possible approach in the first place then such suggestions aren't really helpful.

**Comment by matgis** [comment]
I think this situation only applies to external edits?

Potentially we could fix the issue by applying external changes to code files as replacing the text of the existing node instead of re-creating the node. 🤔

**Comment by j-h-a** [comment]
> I think this situation only applies to external edits?

It is possible to trigger it with the built-in editor as well, but I think there are fewer situations that can trigger it when using the built-in editor. It seems to happen whenever something interferes with processing the `go.property` declarations in a script.

[9682.zip](https://github.com/user-attachments/files/18180759/9682.zip)

You can reproduce it with the built-in editor in the attached minimal project. There is a git repo in the project with one commit, so that it's easy to see when the editor makes changes.

Minimal steps to reproduce:

1. Check git status or open a tool to view uncommitted changes, see that there are none.
2. Open the project in Defold and open `main/number_store.script` in the built-in editor.
3. Add a compilation error to one of the properties, for example add an extra close-bracket to the end of a line: `go.property("number1", 0))`
4. Press save
5. Check git status again and see that the editor has removed overrides of all three properties in all reachable collections

The Defold-editor must keep a cached version of the original in memory, because if you fix the compilation error it puts back the correct values in the collection files on disk. But if you exit Defold after step 5 the data-loss is locked in. When using an external code editor, the invalidation of memory caches may be different, because I haven't noticed if it ever removes and puts the data back. If the collections themselves have also been edited, it becomes difficult to unravel when the unwanted changes are present alongside your deliberate changes.

**Comment by matgis** [comment]
Good points!

So it seems the situation could be improved if treat external changes as text edits, but to fix it completely would require additional effort.
